src/panels/lovelace/cards/hui-calendar-card.ts

```diff
@@ -0,0 +1,178 @@
+import { findEntities } from "../common/find-entites";
+import { applyThemesOnElement } from "../../../common/dom/apply_themes_on_element";
+import "../components/hui-warning";
+import { processConfigEntities } from "../common/process-config-entities";
+import { computeStateName } from "../../../common/entity/compute_state_name";
+import { fetchCalendarEvents } from "../../../data/calendar";
+import { HASSDomEvent } from "../../../common/dom/fire_event";
+import { HA_COLOR_PALETTE } from "../../../common/const";
+
+@customElement("hui-calendar-card")
+export class HuiCalendarCard extends LitElement implements LovelaceCard {
+  public static async getConfigElement(): Promise<LovelaceCardEditor> {
+    await import(
+      /* webpackChunkName: "hui-calendar-card-editor" */ "../editor/config-elements/hui-calendar-card-editor"
+    );
+    return document.createElement("hui-calendar-card-editor");
+  }
+
+  public static getStubConfig(
+    hass: HomeAssistant,
+    entities: string[],
+    entitiesFill: string[]
+  ) {
+    const includeDomains = ["calendar"];
+    const maxEntities = 1;
+    const foundEntities = findEntities(
+      hass,
+      maxEntities,
+      entities,
+      entitiesFill,
+      includeDomains
+    );
+
+    return {
+      entity: foundEntities[0] || "",
+    };
+  }
+
+  @property() public hass?: HomeAssistant;
+
+  @property() private _config?: CalendarCardConfig;
+
+  @property() public _events: CalendarEvent[] = [];
+
+  @property() private _calendars: Calendar[] = [];
+
+  public setConfig(config: CalendarCardConfig): void {
+    if (!config.entities) {
+      throw new Error("Entities must be defined");
+    }
+
+    if (config.entities && !Array.isArray(config.entities)) {
+      throw new Error("Entities need to be an array");
+    }
+
+    this._config = config;
+  }
+
+  public getCardSize(): number {
+    return 4;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass || !this._calendars.length) {
+      return html``;
+    }
+
+    return html`
+      <ha-card>
+        <div class="header">${this._config.title}</div>
+        <ha-full-calendar
+          .events=${this._events}
+          .narrow=${this.offsetWidth < 870}
+          .hass=${this.hass}
+          .views=${["listWeek", "dayGridMonth"]}
+          @view-changed=${this._handleViewChanged}
+        ></ha-full-calendar>
+      </ha-card>
+    `;
+  }
+
+  protected firstUpdated(changedProps: PropertyValues): void {
+    super.firstUpdated(changedProps);
+    const configEntities = this._config!.entities
+      ? processConfigEntities(this._config!.entities)
+      : [];
+
+    const calendars: Calendar[] = [];
+
+    configEntities.forEach((entity, idx) => {
+      calendars.push({
+        entity_id: entity.entity,
+        name: computeStateName(this.hass!.states[entity.entity]),
+        backgroundColor: HA_COLOR_PALETTE[idx % HA_COLOR_PALETTE.length],
+      });
+    });
+
+    this._calendars = calendars;
```

Comment:
This should also go to `setConfig`, if you change something now it is not reflected in the card.